crates/sui-package-resolver/src/lib.rs

```diff
@@ -93,6 +96,32 @@ pub struct Package {
 
 type Linkage = BTreeMap<AccountAddress, AccountAddress>;
 
+#[derive(Clone, Debug)]
+pub enum CleverError {
+    CompleteError {
+        /// The (storage) module ID of the module that the assertion failed in.
+        module_id: ModuleId,
+        /// The name of the error constant.
+        error_identifier: String,
+        /// The value of the error constant. In the case where the error constant is either a
+        /// * A vector of bytes convertible to a valid UTF-8 string; or
+        /// * A numeric value (u8, u16, u32, u64, u128, u256); or
+        /// * A boolean value; of
+        /// * An address value
+        /// Then `Ok(<string_rep_of_constant>)` is returned. Otherwise, `Err(<bytes>)` is returned, and
+        /// the caller is responsible for determining how best to display the error constant.
+        error_constant: std::result::Result<String, Vec<u8>>,
+        /// The line number in the source file where the error occured.
+        source_line_number: u16,
+    },
+    LineNumberOnly {
```

Comment:
nit (optional): avoiding name redundancy 
```suggestion
    LineNumber {
```